### PR TITLE
Add nested contact emails to the work form title tab

### LINKED ISSUE #112: Add contact email field to the title tab of the deposit work form
Both of our early beta users utilize the contact email field (particularly in Acquisitions Serials) as part of their workflow.

### PR DESCRIPTION
**Pull request by aaron-collier**
Closes #112 

Note - this can be merged as is (pending review) but a styling update should happen after #224 to remove the border to more closely match the design.

Nested form:

Screenshot 2024-12-03 at 7 59 32 PM

Show page:

Screenshot 2024-12-03 at 8 04 00 PM

### CODE REVIEW

**Review by justinlittman — CHANGES_REQUESTED** [review]

File: app/forms/contact_email_form.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+# Form for contact emails
+class ContactEmailForm < ApplicationForm
+  attribute :email, :string
+  validates :email, presence: true, format: {
+    with: /\A[^@]+@([a-z0-9-]+\.)+[a-z]{2,4}\z/
+  }, if: :deposit?
```
Comment (justinlittman) [inline app/forms/contact_email_form.rb:8]: I wouldn't make this for deposit only. It should be an error for draft as well.

File: app/services/cocina_description_support.rb [hunk 2]
```diff
@@ -27,6 +27,19 @@ def self.title(title:)
     [{ value: title }]
   end
 
+  def self.contact_emails(contact_emails:)
+    contact_emails.map do |contact_email|
+      contact_email = contact_email.attributes if contact_email.respond_to?(:attributes)
```
Comment (justinlittman) [inline app/services/cocina_description_support.rb:32]: Why is this necessary? (Can you put explanation in a code comment.)

File: app/views/works/show.html.erb [hunk 3]
```diff
@@ -6,8 +6,9 @@
   <%= render Elements::SpinnerComponent.new %>
 <% end %>
 
-<%= render Elements::Tables::TableComponent.new(id: 'title-table', classes: 'table-show mb-5', label: 'Title') do |component| %>
+<%= render Elements::Tables::TableComponent.new(id: 'title-table', classes: 'table-work mb-5', label: 'Title and contact') do |component| %>
```
Comment (justinlittman) [inline app/views/works/show.html.erb:9]: I don't think the table-show class should have changed.

File: db/structure.sql [hunk 4]
```diff
@@ -158,6 +158,39 @@ CREATE SEQUENCE public.collections_id_seq
 ALTER SEQUENCE public.collections_id_seq OWNED BY public.collections.id;
 
 
+--
```
Comment (justinlittman) [inline db/structure.sql:161]: Is this db change cruft?

File: spec/system/show_work_spec.rb [hunk 5]
```diff
@@ -145,9 +145,11 @@
 
     # Title table
     within('table#title-table') do
-      expect(page).to have_css('caption', text: 'Title')
+      expect(page).to have_css('caption', text: 'Title and contact')
```
Comment (justinlittman) [inline spec/system/show_work_spec.rb:148]: Is the discrepancy between "Title and contact" and "Title & contact" deliberate?

**Review by justinlittman — CHANGES_REQUESTED** [review]

File: spec/system/show_work_spec.rb (hunk 5 above)
Comment (aaron-collier) [reply spec/system/show_work_spec.rb:148]: I assume you mean between the tab label and the show page? Yes, they are different in the figma designs.

File: db/structure.sql (hunk 4 above)
Comment (aaron-collier) [reply db/structure.sql:161]: Must be, I'll remove.

File: app/views/works/show.html.erb (hunk 3 above)
Comment (aaron-collier) [reply app/views/works/show.html.erb:9]: I don't think that was deliberate, maybe part of rebasing. I'll fix. Thanks for catching this.

File: app/services/cocina_description_support.rb (hunk 2 above)
Comment (aaron-collier) [reply app/services/cocina_description_support.rb:32]: Hmm, this basically directly from the related_links/related_works nested components. I'll see if I can determine if it's necessary or not.

Comment (aaron-collier) [reply app/services/cocina_description_support.rb:34]: done.

**Review by justinlittman — APPROVED** [review]
